xbmc/addons/AddonInfo.h

```diff
@@ -107,13 +222,18 @@ namespace ADDON
 
     const std::string& ID() const { return m_id; }
     TYPE MainType() const { return m_mainType; }
+    bool IsType(TYPE type) const;
+    const std::vector<CAddonType>& Types() const { return m_types; }
+    const CAddonType* Type(TYPE type) const;
+    std::string MainLibPath() const;
+
     const AddonVersion& Version() const { return m_version; }
     const AddonVersion& MinVersion() const { return m_minversion; }
     const std::string& Name() const { return m_name; }
     const std::string& License() const { return m_license; }
     const std::string& Summary() const { return m_summary; }
     const std::string& Description() const { return m_description; }
-    const std::string& LibName() const { return m_libname; }
+    const std::string& LibName() const { return m_types[0].m_libname; }
```

Comment:
Here it is needed in future for binary addons. I think to change the name to `MainLibName()`.